test/integration/targets/zabbix_host/tasks/zabbix_host_tests.yml

```diff
@@ -0,0 +1,662 @@
+
+- name: expect to succeed and that things changed
+  assert:
+      that:
+          - "zabbix_host1.changed"
+
+- name: "test: change description (again)"
+  local_action:
+    module: zabbix_host
+    server_url: "{{ zabbix_server_url }}"
+    login_user: "{{ zabbix_login_user }}"
+    login_password: "{{ zabbix_login_password }}"
+    host_name: ExampleHost
+    description: "My ExampleHost Description Changed"
+  register: zabbix_host1
+
+- name: updating with same values should be idempotent
+  assert:
+      that:
+          - "not zabbix_host1.changed"
+
+- name: "test: change host groups (adding one group)"
+  local_action:
+    module: zabbix_host
+    server_url: "{{ zabbix_server_url }}"
+    login_user: "{{ zabbix_login_user }}"
+    login_password: "{{ zabbix_login_password }}"
+    host_name: ExampleHost
+    host_groups:
+      - Linux servers
+      - Zabbix servers
+      - Virtual machines
+  register: zabbix_host1
+
+- name: expect to succeed and that things changed
+  assert:
+      that:
+          - "zabbix_host1.changed"
+
+- name: "test: change host groups (remove one group)"
+  local_action:
+    module: zabbix_host
+    server_url: "{{ zabbix_server_url }}"
+    login_user: "{{ zabbix_login_user }}"
+    login_password: "{{ zabbix_login_password }}"
+    host_name: ExampleHost
+    host_groups:
+      - Linux servers
+      - Zabbix servers
+  register: zabbix_host1
+
+- name: expect to succeed and that things changed
+  assert:
+      that:
+          - "zabbix_host1.changed"
+
+- name: "test: change host groups (add one group using force=no)"
+  local_action:
+    module: zabbix_host
+    server_url: "{{ zabbix_server_url }}"
+    login_user: "{{ zabbix_login_user }}"
+    login_password: "{{ zabbix_login_password }}"
+    host_name: ExampleHost
+    host_groups:
+      - Virtual machines
+    force: no
+  register: zabbix_host1
+
+- name: expect to succeed and that things changed
+  assert:
+      that:
+          - "zabbix_host1.changed"
+
+
+- name: "test: change host groups (check whether we are at three groups)"
+  local_action:
+    module: zabbix_host
+    server_url: "{{ zabbix_server_url }}"
+    login_user: "{{ zabbix_login_user }}"
+    login_password: "{{ zabbix_login_password }}"
+    host_name: ExampleHost
+    host_groups:
+      - Linux servers
+      - Zabbix servers
+      - Virtual machines
+  register: zabbix_host1
+
+- name: expect to succeed and that things have not changed
+  assert:
+      that:
+          - "not zabbix_host1.changed"
+
+- name: "test: change host groups (attempt to remove all host groups)"
+  local_action:
+    module: zabbix_host
+    server_url: "{{ zabbix_server_url }}"
+    login_user: "{{ zabbix_login_user }}"
+    login_password: "{{ zabbix_login_password }}"
+    host_name: ExampleHost
+    host_groups:
+        -
+  register: zabbix_host1
+  ignore_errors: yes
+
+- name: expect to fail
+  assert:
+      that:
+          - "zabbix_host1|failed"
```

Comment:
Based on https://github.com/ansible/proposals/issues/83 this should now read `zabbix_host1 is failed`